src/main/kotlin/net/minecrell/pluginyml/paper/PaperPluginDescription.kt

```diff
@@ -56,28 +57,34 @@ class PaperPluginDescription(project: Project) : PluginDescription() {
     @Input @Optional @JsonProperty("has-open-classloader") var hasOpenClassloader: Boolean? = null
     @Input @Optional @JsonProperty("folia-supported") var foliaSupported: Boolean? = null
 
-    @Nested @Optional @JsonProperty("dependencies") @JsonSerialize(`as` = Collection::class)
-    var depends: NamedDomainObjectContainer<DependencyDefinition> = project.container(DependencyDefinition::class.java)
-    @Nested @Optional @JsonProperty("load-before") @JsonSerialize(`as` = Collection::class)
-    var loadBefore: NamedDomainObjectContainer<LoadDefinition> = project.container(LoadDefinition::class.java)
-    @Nested @Optional @JsonProperty("load-after") @JsonSerialize(`as` = Collection::class)
-    var loadAfter: NamedDomainObjectContainer<LoadDefinition> = project.container(LoadDefinition::class.java)
+    @Nested @Optional @JsonIgnore
+    var serverDependencies: NamedDomainObjectContainer<DependencyDefinition> = project.container(DependencyDefinition::class.java)
+    @Nested @Optional @JsonIgnore
+    var bootstrapDependencies: NamedDomainObjectContainer<DependencyDefinition> = project.container(DependencyDefinition::class.java)
+
+    @JsonProperty("dependencies") @JsonSerialize(contentAs = Collection::class)
```

Comment:
It still needs some sort of annotation since it's a getter function. Maybe `@JsonGetter` is the correct thing to use? It seems to be able to infer the name of a field based on the function.

Reply:
8ac648d63fa26185b871b09f4cfa6a3712d48436